### PR TITLE
Update snippets to compile in .NET 7

### LINKED ISSUE #33286: Sample code from MS docs should ALWAYS compile without errors and warnings
The sample code in this article generates CS8618. If Microsoft is serious about addressing nullability issues, then the sample code should be check in the current versions of the language with default nullability settings.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: d2822c97-72b4-3514-a07c-c1477114f410
* Version Independent ID: 3f9a9b3f-bbac-d46f-ef0b-d0b5048d463b
* Content: [How to: Explicitly Throw Exceptions](https://learn.microsoft.com/en-us/dotnet/standard/exceptions/how-to-explicitly-throw-exceptions)
* Content Source: [docs/standard/exceptions/how-to-explicitly-throw-exceptions.md](https://github.com/dotnet/docs/blob/main/docs/standard/exceptions/how-to-explicitly-throw-exceptions.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

---
[Associated WorkItem - 62393](https://dev.azure.com/msft-skilling/Content/_workitems/edit/62393)

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #33286.

image

### CODE REVIEW

File: samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
     {
-        FileStream fs;
-        try
+        public static void Main()
         {
-            // Opens a text tile.
-            fs = new FileStream(@"C:\temp\data.txt", FileMode.Open);
-            var sr = new StreamReader(fs);
+            // <Snippet1>
+            FileStream? fs = null;
```
Comment (gewarren) [inline samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs:8]: Make `fs` nullable and set it to null.

File: samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-        {
-            Console.WriteLine($"[Data File Missing] {e}");
-            throw new FileNotFoundException(@"[data.txt not in c:\temp directory]", e);
-        }
-        finally
-        {
-            if (fs != null)
-                fs.Close();
+                // A value is read from the file and output to the console.
+                string? line = sr.ReadLine();
```
Comment (gewarren) [inline samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs:16]: Make `line` nullable.

File: samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs [hunk 3]
```diff
@@ -1,32 +1,32 @@
-﻿// <Snippet1>
-using System;
-using System.IO;
-
-public class ProcessFile
+﻿namespace Example
```
Comment (gewarren) [inline samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs:1]: Add a namespace.

File: samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
     {
-        FileStream fs;
-        try
+        public static void Main()
         {
-            // Opens a text tile.
-            fs = new FileStream(@"C:\temp\data.txt", FileMode.Open);
-            var sr = new StreamReader(fs);
+            // <Snippet1>
+            FileStream? fs = null;
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR/Exception.Throwing/CS/throw.cs:11]: Being a fan of `var`, I'd consider this alternative:

```suggestion
            var fs = default(FileStream);
```

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

I have one alternative suggestion. I don't feel strongly about either syntax.

You can :shipit: when ready.

**Pull request merged**